File: exam-system-java/common/src/main/java/com/yyjy/exam/common/constant/CacheConstant.java
package com.yyjy.exam.common.constant;

public final class CacheConstant {

    private CacheConstant() {
    }

    public static final String EXAM_RECORD_CACHE = "exam_record:";
    public static final String QUESTION_DETAIL_KEY = "question:detail:";
    public static final String QUESTION_CATEGORY_KEY = "question:category:";
    public static final String PAPER_DETAIL_KEY = "paper:detail:";
    public static final String EXAM_RECORD_DETAIL_KEY = "exam_record:detail:";
    public static final String POPULAR_QUESTIONS_KEY = "question:popular:";
    public static final String QUESTION_VIEW_COUNT_KEY = "question:view_count:";
    public static final int POPULAR_QUESTIONS_COUNT = 10;
    public static final long DEFAULT_EXPIRE_SECONDS = 1800;
    public static final long HOT_DATA_EXPIRE_SECONDS = 3600;
    public static final String CAPTCHA_REGISTER_KEY = "captcha:register:";
    public static final String DASHBOARD_STATS_KEY = "dashboard:stats";
}

File: exam-system-server/src/main/kotlin/com/yyjy/constants/CacheConstant.kt
package com.yyjy.constants

/**
 * 缓存常量类
 * 定义系统中使用的缓存名称和key前缀
 */
object CacheConstant {
    /**
     * 题目模块缓存名称
     */
    const val QUESTION_CACHE: String = "question"

    /**
     * 试卷模块缓存名称
     */
    const val PAPER_CACHE: String = "paper"

    /**
     * 考试记录模块缓存名称
     */
    const val EXAM_RECORD_CACHE: String = "exam_record"

    /**
     * 题目详情缓存key前缀
     */
    const val QUESTION_DETAIL_KEY: String = "question:detail:"

    /**
     * 分类题目列表缓存key前缀
     */
    const val QUESTION_CATEGORY_KEY: String = "question:category:"

    /**
     * 试卷详情缓存key前缀
     */
    const val PAPER_DETAIL_KEY: String = "paper:detail:"

    /**
     * 考试记录详情缓存key前缀
     */
    const val EXAM_RECORD_DETAIL_KEY: String = "exam_record:detail:"

    /**
     * 热门题目缓存key
     */
    const val POPULAR_QUESTIONS_KEY: String = "question:popular"

    /**
     * 题目访问计数key
     */
    const val QUESTION_VIEW_COUNT_KEY: String = "question:view_count"

    /**
     * 热门题目数量
     */
    const val POPULAR_QUESTIONS_COUNT: Int = 10

    /**
     * 缓存过期时间（秒）
     */
    const val DEFAULT_EXPIRE_SECONDS: Long = 1800 // 30分钟

    /**
     * 热点数据缓存过期时间（秒）
     */
    const val HOT_DATA_EXPIRE_SECONDS: Long = 3600 // 1小时
}